commands/remove_test.go

```diff
@@ -4,28 +4,73 @@
 package commands
 
 import (
+	"io/ioutil"
 	"net/http"
+	"os"
 	"testing"
 
 	"github.com/openfaas/faas-cli/test"
 )
 
+const testStack = `
+provider:
+  name: faas
+  gateway: http://127.0.0.1:8080
+
+functions:
+  fn1:
+    lang: go
+    handler: ./fn1
+`
+
 func Test_remove(t *testing.T) {
 	s := test.MockHttpServer(t, []test.Request{
 		{
 			Method:             http.MethodDelete,
 			Uri:                "/system/functions",
 			ResponseStatusCode: http.StatusOK,
 		},
+		{
+			Method:             http.MethodDelete,
+			Uri:                "/system/functions",
+			ResponseStatusCode: http.StatusOK,
+		},
 	})
 	defer s.Close()
 
+	tmpfile, err := ioutil.TempFile("", "stack.*.yml")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer os.Remove(tmpfile.Name())
+
+	if _, err := tmpfile.Write([]byte(testStack)); err != nil {
+		tmpfile.Close()
+		t.Fatal(err)
+	}
+	if err := tmpfile.Close(); err != nil {
+		t.Fatal(err)
+	}
+
 	resetForTest()
 
 	faasCmd.SetArgs([]string{
 		"remove",
 		"--gateway=" + s.URL,
 		"test-function",
 	})
-	faasCmd.Execute()
+	stdoutWithoutYaml := test.CaptureStdout(func() { faasCmd.Execute() })
+
+	faasCmd.SetArgs([]string{
+		"remove",
+		"--yaml=" + tmpfile.Name(),
+		"--gateway=" + s.URL,
+		"test-function",
+	})
+	stdoutWithYaml := test.CaptureStdout(func() { faasCmd.Execute() })
+
+	if stdoutWithYaml != stdoutWithoutYaml {
```

Comment:
I'm confused with what's happening here, please can you make the test clearer and also explain it for me?

Reply:
I first create a stack.yml file. This was always used as soon as it existed before this change.
Now I basically only call 'remove' 2 times.
Once with -yaml=something and once without.

I compare both outputs and if they are the same
both times the yaml file was used. The test then fails because we don't want to use the yaml file anymore if a function to delete is explicitly specified.

Reply:
I didn't get that by eyeballing it. Is there another way you could write the test?

Reply:
I hope it's easier to understand now.